src/storage/wal_replay.cpp

```diff
@@ -360,37 +344,39 @@ void WriteAheadLogDeserializer::ReplayEntry(WALType entry_type) {
 //===--------------------------------------------------------------------===//
 // Replay Version
 //===--------------------------------------------------------------------===//
-void WriteAheadLogDeserializer::ReplayVersion() {
+std::unique_ptr<ParseInfo> WriteAheadLogDeserializer::ReplayVersion() {
 	state.wal_version = deserializer.ReadProperty<idx_t>(101, "version");
+	auto info = make_uniq<ParseInfo>(ParseInfoType::VERSION_INFO);
+	return info;
```

Comment:
I don't understand the benefit of returning an empty ParseInfo here, don't we want to add the `state.wal_version` to this?

Reply:
Returning `nullptr` causes the `WALReader` to terminate the loop prematurely. It's interpreted as "no more WAL records" left. I didn't return the version because `WALReader` can get it from `ReplayState`.